### PR TITLE
Icon color

### PR DESCRIPTION
**Pull request by divyanshiGupta**

### CODE REVIEW

**Comment by joshuawilson** [comment]
What is the issue that this is being done for?

**Comment by divyanshiGupta** [comment]
Support for color of work item status icons is being added here.

**Comment by joshuawilson** [comment]
Can you please point to github or OSIO issue where it has been discussed? I don't have any context to know why this is being changed.

**Comment by divyanshiGupta** [comment]
screenshot_20170928-193836 01

**Comment by divyanshiGupta** [comment]
As per the ux there was a need to add color to the work item status icons.

**Comment by joshuawilson** [comment]
Is there no issue this was listed on?

File: src/app/app.module.ts [hunk 1]
```diff
@@ -5,7 +5,6 @@ import { BrowserModule } from '@angular/platform-browser';
 import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
 import { FormsModule } from '@angular/forms';
 import { HttpModule } from '@angular/http';
-
```
Comment (sanbornsen) [inline src/app/app.module.ts]: Why was this line removed? Good practice to keep angular imports in separate section.

File: src/app/icon/almicon.directive.ts [hunk 2]
```diff
@@ -39,11 +39,15 @@ export class AlmIconDirective implements OnInit, OnChanges {
     });
 
     if (this.iconType in IconMap) {
-      IconMap[this.iconType].forEach((item: any) => {
+      this.iconColor = IconMap[this.iconType].colour;
+      element.setAttribute("style","color:"+this.iconColor);
```
Comment (sanbornsen) [inline src/app/icon/almicon.directive.ts]: * Space after ','
* Space before and after '+'

File: src/app/icon/almicon.directive.ts [hunk 3]
```diff
@@ -39,11 +39,15 @@ export class AlmIconDirective implements OnInit, OnChanges {
     });
 
     if (this.iconType in IconMap) {
-      IconMap[this.iconType].forEach((item: any) => {
+      this.iconColor = IconMap[this.iconType].colour;
```
Comment (sanbornsen) [inline src/app/icon/almicon.directive.ts]: We are using the american spelling of 'colour' everywhere. To keep it consistent let's use 'color' instead of 'colour'

File: src/app/icon/almicon.directive.ts [hunk 4]
```diff
@@ -8,7 +8,7 @@ import { IconMap } from './iconmap';
 export class AlmIconDirective implements OnInit, OnChanges {
   @Input() 
   iconType: string = 'none';
-
+  iconColor: string = 'none';
```
Comment (sanbornsen) [inline src/app/icon/almicon.directive.ts]: Why do you need a class variable **iconColor** ? You are just using it in **addIcon** function. It can be a local variable, can't it?

**Comment by joshuawilson** [comment]
@divyanshiGupta can you please put a link to an issue, task, or bug that defines why this needs to be done?

**Comment by divyanshiGupta** [comment]
@joshuawilson as per the ux team this fix was needed but there was no issue raised as such but I have asked the ux team to raise this issue and link it to this PR.

**Comment by nimishamukherjee** [comment]
@joshuawilson please refer to https://github.com/openshiftio/openshift.io/issues/325
You can have a look at the Invision link added there - https://redhat.invisionapp.com/share/YZDOM6S2R#/screens
This is how the list view will look.
This PR by @divyanshiGupta addresses the state visuals in particular.

**Comment by mindreeper2420** [comment]
Wouldn’t it be more extendable to just add classes here, rather than hard code color codes into the .ts file? That way CSS is controlling the colors, color variables can be used, and the classes can be used to make additional changes in the future (size, color, etc.).

**Comment by divyanshiGupta** [comment]
@mindreeper2420 yes I tried using classes but as we are using patternfly color pallette the color codes are defined as variables in the library and thus I decided to make corresponding color classes and use them instead but as the icon module didn't have a corresponding stylesheet so after consulting with @nimishamukherjee and @vikram-raj about where should the classes be placed,  we decided to put the classes under assets folder in custom stylesheet but it didn't worked out as planner was not able to find the classes. So for the time being color codes are being used. I am working on including the classes instead.

**Comment by joshuawilson** [comment]
@mindreeper2420 I'm going to merge this, if you want this changed please create a new issue with what you want done.
